src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLSPTextViewConnectionListener.cs

```diff
@@ -111,31 +113,36 @@ public void SubjectBuffersConnected(ITextView textView, ConnectionReason reason,
             // Initialize the user's options and start listening for changes.
             // We only want to attach the option changed event once so we don't receive multiple
             // notifications if there is more than one TextView active.
-            if (_razorEditorOptions == null && textView.TextBuffer.IsRazorLSPBuffer())
+            if (!textView.TextBuffer.Properties.ContainsProperty(RazorEditorOptions) &&
+                textView.TextBuffer.IsRazorLSPBuffer())
             {
-                _razorEditorOptions = _editorOptionsFactory.GetOptions(textView);
-                Assumes.Present(_razorEditorOptions);
+                var razorEditorOptions = _editorOptionsFactory.GetOptions(textView);
```

Comment:
Hmmmm I just realized we're getting options for the `ITextView` not the buffer here, I worry that might misbehave in scenarios with multiple `ITextView`s.

Could you try opening the same document in 2 text views and changing the setting in one and see if it reflects in the other?